agent/acs/handler/attach_eni_handler.go

```diff
@@ -85,64 +85,67 @@ func (attachENIHandler *attachENIHandler) handleMessages() {
 	for {
 		select {
 		case message := <-attachENIHandler.messageBuffer:
-			attachENIHandler.handleSingleMessage(message)
+			if err := attachENIHandler.handleSingleMessage(message); err != nil {
+				seelog.Warnf("Unable to handle ENI Attachment message [%s]: %v", message.String(), err)
+			}
 		case <-attachENIHandler.ctx.Done():
 			return
 		}
 	}
 }
 
 // handleSingleMessage acks the message received
-// TODO: Send response to validate ENI attachment
-func (attachENIHandler *attachENIHandler) handleSingleMessage(message *ecsacs.AttachTaskNetworkInterfacesMessage) error {
+func (handler *attachENIHandler) handleSingleMessage(message *ecsacs.AttachTaskNetworkInterfacesMessage) error {
+	receivedAt := time.Now()
 	// Validate fields in the message
-	err := validateAttachTaskNetworkInterfacesMessage(message)
-	if err != nil {
-		return errors.Wrapf(err, "attach eni message handler: error validating AttachTaskNetworkInterfac message received from ECS")
+	if err := validateAttachTaskNetworkInterfacesMessage(message); err != nil {
+		return errors.Wrapf(err,
+			"attach eni message handler: error validating AttachTaskNetworkInterface message received from ECS")
 	}
-
-	err = attachENIHandler.acsClient.MakeRequest(&ecsacs.AckRequest{
+	if err := handler.acsClient.MakeRequest(&ecsacs.AckRequest{
 		Cluster:           message.ClusterArn,
 		ContainerInstance: message.ContainerInstanceArn,
 		MessageId:         message.MessageId,
-	})
-	if err != nil {
+	}); err != nil {
 		seelog.Warnf("Failed to ack request with messageId: %s, error: %v", aws.StringValue(message.MessageId), err)
 	}
-
 	// Check if this is a duplicate message
-	eniMac := aws.StringValue(message.ElasticNetworkInterfaces[0].MacAddress)
-	if _, ok := attachENIHandler.state.ENIByMac(eniMac); ok {
-		seelog.Infof("Duplicate ENI attachment message for ENI with MAC address: %s", eniMac)
-		return nil
+	mac := aws.StringValue(message.ElasticNetworkInterfaces[0].MacAddress)
+	if eniAttachment, ok := handler.state.ENIByMac(mac); ok {
+		seelog.Infof("Duplicate ENI attachment message for ENI with MAC address: %s", mac)
+		eniAckTimeoutHandler := ackTimeoutHandler{mac: mac, state: handler.state}
+		return eniAttachment.StartTimer(eniAckTimeoutHandler.handle)
```

Comment:
Are you asking if it's ok to get rid of this line? We need to make sure that the expiration timer starts if the Agent was restarted, which is why I added this line.